src/components/optimizerbox/Optimizerbox.js

```diff
@@ -1,39 +1,168 @@
+        };
+
+    // Log
+    console.log('Sending data to optimizer:', optimizerData);
+
+    try {
+      // POST request
+      const response = await fetch('http://localhost:8080/optimization', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(optimizerData),
+      });
+
+      if (response.ok) {
+        const data = await response.json();
+
+
+        // Add new result
+        addOptimizationResult(data);
+
+        // Store output
+        setOutputParameters({
+          directAdType: data.directAdType,
+          directAdFrequency: data.directAdFrequency,
+          trainChiefsFrequency: data.trainChiefsFrequency,
+          directAdNrOfVillages: data.directAdNrOfVillages,
+          trainChiefsNumber: data.trainChiefsNumber,
+          avgAdopters: data.avgAdopters,
+          nrOfDirectAds: data.nrOfDirectAds,
+          nrOfChiefTrainings: data.nrOfChiefTrainings,
+          totalCost: data.totalCost,
+        });
+      } else {
+        console.error('Error:', response.status, response.statusText);
+      }
+    } catch (error) {
+      console.error('Fetch Error:', error.message, error.stack);
+    }
+  };
+
+  
+  return (
+    <div className="optimizerBox">
+      <h2>The Optimizer</h2>
+      <div className="flexContainer">
+        <div className="inputGroup">
+          <Dropdown label= "Optimization Type" value={optimizationType} setValue={setOptimizationType}>
+            <option disabled value="">
+                Please choose a type
+              </option>
+            <option value="maxAdopters">Max Adopters</option>
+            <option value="maxKnowledge">Max Knowledge</option>
+            <option value="minCost">Min Costs</option>
+            <option value="test">Test</option>
+            </Dropdown>
         </div>
-    );
-};
+        <div className="inputGroup">
+          <label>Additional Constraints:</label>
+          <TextInput label= "Budget" value={budget} setValue={setBudget} />
+          <TextInput label= "Fixed Costs for Direct Ad" value={fixedCostsDirectAd} setValue={setFixedCostsDirectAd} />
+          <TextInput label= "Fixed Costs for Training of Chiefs" value={fixedCostsTrainChiefs} setValue={setFixedCostsTrainChiefs} />
+          <TextInput label= "Variable Costs for Direct Ad" value={variableCostsDirectAd} setValue={setVariableCostsDirectAd} />
+          <TextInput label= "Variable Costs for Discount" value={variableCostsDiscount} setValue={setVariableCostsDiscount} />
+          <TextInput label= "Variable Costs for Delayed" value={variableCostsDelayed} setValue={setVariableCostsDelayed} />
+          <TextInput label= "Variable Costs for Delayed Discount" value={variableCostsDelayedDiscount} setValue={setVariableCostsDelayedDiscount} />
+          <TextInput label= "Variable Costs for Training of Chiefs" value={variableCostsTrainChiefs} setValue={setVariableCostsTrainChiefs} />
```

Comment:
aus irgendeinem Grund kann man in dem Textfeld immer nur einen character bearbeiten (reinschreiben, rauslöschen). Danach verschwindet der Cursor und man muss neu ins Textfeld reinklicken. Ich verstehe gerade nicht, wo das Problem liegt. Die Modelbox ist ja ähnlich implementiert..